app/models/Post.php

```diff
@@ -0,0 +1,127 @@
+<?php
+namespace app\models;
+class Post extends Entity {
+
+    /** Model Add a post
+     * @param $title
+     * @param $body
+     * @param $userId
+     * @return null
+     */
+    public function addPost($title, $body, $userId) {
+        // Prepare the SQL query to insert a new blog post
+        $sql = "INSERT INTO Posts (title, body, user_id, created_at) 
+                VALUES (:title, :body, :userId, NOW())";
+
+        // Prepare the statement
+        $this->db->query($sql);
+
+        // Bind the parameters to prevent SQL injection
+        $this->db->bind(':title', $title);
+        $this->db->bind(':body', $body);
+        $this->db->bind(':userId', $userId);
+
+        // Execute the query
+        return $this->db->execute();
+    }
+
+    /** Model Update a post
+     * @param $id
+     * @param $title
+     * @param $body
+     * @return null
+     */
+    public function updatePost($id, $title, $body) {
+        $sql = "UPDATE Posts SET title = :title, body = :body, updated_at = NOW() WHERE id = :id";
+        $this->db->query($sql);
+        $this->db->bind(':title', $title);
+        $this->db->bind(':body', $body);
+        $this->db->bind(':id', $id);
+        return $this->db->execute();
+    }
+
+
+    /** Model Get all the posts
+     * @return mixed
+     */
+    // Retrieve all blog posts from the database
+    public function getPosts() {
+        $this->db->query("SELECT * FROM Posts ORDER BY created_at DESC");
+        return $this->db->resultSet();
+    }
+
+    /** Model Retrieve a single blog post by ID
+     * @param $id
+     * @return mixed
+     */
+    public function getPostByID($id) {
+        $this->db->query("SELECT * FROM Posts WHERE id = :id");
+        $this->db->bind(':id', $id);
+        return $this->db->single();
+    }
+
+    /** Model Increasing the comment count
+     * @param $postId
+     * @return void
+     */
+    public function incrementCommentCount($postId) {
+        $this->db->query("UPDATE Posts Set comment_count = comment_count + 1 WHERE id= :id");
```

Comment:
Just confirming that you're going to stick with this approach, instead of using `select count(*) from comments where post_id = $id`?

Reply:
Haha, I would go about changing this logic, because I spent huge time messing around with front-endish thing with how to render the comment count, I just became lazy to change it especially when you told me we would dump the whole thing once done =.=, never mind, I would refactor to use the optimized way.